AsyncDisplayKitTests/ASDisplayNodeTests.m

```diff
@@ -1964,6 +1964,23 @@ - (void)testThatLayerBackedSubnodesAreMarkedInvisibleBeforeDeallocWhenSupernodes
   XCTAssertNoThrow([nodeView removeFromSuperview]);
 }
 
+/**
+ * CALayer.delegate is actually `assign` not weak. This test demonstrates how the layer.delegate
+ * pointer dangling can cause crashes.
+ */
+- (void)testThatLayerDelegateDoesntDangleAndCauseCrash
+{
+  NS_VALID_UNTIL_END_OF_SCOPE UIView *host = [[UIView alloc] init];
+
+  __block NS_VALID_UNTIL_END_OF_SCOPE ASDisplayNode *node = [[ASDisplayNode alloc] init];
+  node.layerBacked = YES;
+
+  [host addSubnode:node];
+  [self executeOffThread:^{
+    node = nil;
```

Comment:
Could you comment this code more?  I think the idea is:

// Running on main thread
// Cause retain count of node to fall to zero _synchronously on a background thread_ (pausing main thread)
// ASDealloc2MainObject queues actual call to -dealloc to occur on the main thread
// Continue execution on main, before the dealloc can run, to dealloc the host view
// Node is in an invalid state (about to dealloc, not valid to retain) but accesses to sublayer delegates causes attempted retain — unless weak variable works correctly